SETUP.md

```diff
@@ -1,148 +1,128 @@
+- [Yarn](https://yarnpkg.com/) - JavaScript package manager
+- [Redis](https://redis.io/) - Data structure store
+- [ImageMagick](https://imagemagick.org/) - Image manipulation library
+
+#### Cloning From GitHub
 To clone the repository, either use the Git GUI if you have one installed or enter the following commands:
 ```sh
 git clone https://github.com/CircuitVerse/CircuitVerse.git
 cd CircuitVerse
 ```
-**Note:** If you want to contribute, first fork the original repository and clone the **forked** repository into your local machine. If you don't do this, you will not be able to make commits or change any files.
+**Note:** If you want to contribute, first fork the original repository and clone your **forked** repository into your local machine. If you don't do this, you will not be able to make commits or change any files.
 ```sh
 git clone https://github.com/<username>/CircuitVerse.git
 cd CircuitVerse
 ```
 
-## Setup Instructions
-Please go through the [Contribution Guidelines](CONTRIBUTING.md) before going forward with any development. This helps us keep the process streamlined and results in better PRs.
-You will also need the required software, detailed above. `Redis` and `PostgreSQL` should be running.
-
-**Note:** You might want to use the docker instructions if you do not want to setup your own environment.
-
-1. Install ruby using RVM, use ruby-2.6.5
-1. Install bundler : `gem install bundler`
-1. Install dependencies: `bundle install`
-1. Install Yarn dependencies: `yarn`
-1. Configure your PostgreSQL database in `config/database.yml` (copy `config/database.example.yml` for the template)
+#### Setup
+1. Install Ruby bundler : `gem install bundler`
+2. Install Ruby dependencies: `bundle install`
+3. Install Yarn dependencies: `yarn`
+4. Configure your PostgreSQL database in `config/database.yml` (copy `config/database.example.yml` for the template)
      * **Note:** The Postgres credentials need to be updated to your currently running database
-1. Create database: `rails db:create`
-1. Run migrations: `rails db:migrate`
-1. Start Sidekiq: `bundle exec sidekiq -e development -q default -q mailers -d -L tmp/sidekiq.log` (currently in development)
+5. Create database: `rails db:create`
+6. Run database migrations: `rails db:migrate`
+7. Start Sidekiq (background processes & job queue): `bundle exec sidekiq -e development -q default -q mailers -d -L tmp/sidekiq.log`
 
 Then, local development can be started with `rails s -b 127.0.0.1 -p 8080`. Navigate to `127.0.0.1:8080` in your web browser to access the website.
 
-### CircuitVerse API setup instructions
-CircuitVerse API uses `RSASSA` cryptographic signing that requires `private` and associated `public` key. To generate the keys RUN the following commands in `CircuitVerse/`
-```
-openssl genrsa -out config/private.pem 2048
-openssl rsa -in config/private.pem -outform PEM -pubout -out config/public.pem
-```
+#### Additional instructions for Ubuntu
```

Comment:
Can we add additional instructions for windows to install redis? Redis isn't directly runnable on windows and there is more of a hacky way to do it. Refer this:https://stackoverflow.com/questions/6476945/how-do-i-run-redis-on-windows/19579610#19579610 (the following answer works well)